clear-smoke.lua

```diff
@@ -1,25 +1,28 @@
--- Removes all smoke from the map
-
---[====[
-
-clear-smoke
-===========
-
-Removes all smoke from the map. Note that this can leak memory and should be
-used sparingly.
-
-]====]
+local flows_to_delete = {}
 
 function clearSmoke(flows)
+    local success
     for i = #flows - 1, 0, -1 do
-        if flows[i].type == df.flow_type.Smoke then
+        local flow = flows[i]
+        if flow.type == df.flow_type.Smoke then
             flows:erase(i)
+            flows_to_delete[flow] = true
```

Comment:
why can't this be deleted inline? other tools successfully do it that way, e.g. https://github.com/DFHack/dfhack/blob/develop/plugins/lua/dwarfvet.lua#L129-L137